packages/types/src/payment/provider.ts

```diff
@@ -29,7 +32,7 @@ export type PaymentProviderContext = {
   /**
    * The customer associated with this payment.
    */
-  customer?: Record<string, unknown> // TODO: type
+  customer?: CustomerDTO
```

Comment:
suggestion: The payment should have his own address and customer DTO to define what is expected from those DTO, they are not necessarely entire custom or address no? here it looks like we need to provide all proprs from those DTO that are coming from another domain, is that true?

Reply:
Yep agree, I created separate DTOs which are, for now, just partials from the original DTOs.